lib/Code/TidyAll/Plugin.pm

```diff
@@ -28,10 +28,12 @@ has 'tidyall'            => ( is => 'ro', required => 1, weak_ref => 1 );
 has 'weight'             => ( is => 'lazy' );
 
 # Internal
-has 'ignore_regex' => ( is => 'lazy' );
-has 'ignores'      => ( is => 'lazy' );
-has 'select_regex' => ( is => 'lazy' );
-has 'selects'      => ( is => 'lazy' );
+has 'global_ignore_regex' => ( is => 'lazy' );
+has 'global_ignores'      => ( is => 'lazy' );
```

Comment:
Instead of making every plugin generate these two attributes over and over let's just move the regex to `Code::TidyAll` and then call `$self->tidyall->global_ignore_regex` in the `matches_path` method.